Add Image Converter to community-plugins.json

xRyul opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin:
https://github.com/xRyul/obsidian-image-converter/

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
Please remove the build folder from the repo and add it to your .gitignore - compiled release files shouldn't get mixed with source code in your commits and should go to the releases files instead.

[import * as path from 'path';](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L2)
These imports are only available from a nodeJS runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[MyPluginSettings](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L4)
[MyPlugin](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L32)
[SampleSettingTab](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L548)
Rename these

[newPath = path.join(newPath, newName);](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L107)
Just join the two with a slash `/`, no need to import `path`

[console.log('replace text', linkText, newLinkText);](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L115)
Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.

[.setName('Image Resize Mode')](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L591)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

xRyul commented:
> Please remove the build folder from the repo and add it to your .gitignore - compiled release files shouldn't get mixed with source code in your commits and should go to the releases files instead.
> 
> [import * as path from 'path';](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L2) These imports are only available from a nodeJS runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.
> 
> [MyPluginSettings](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L4) [MyPlugin](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L32) [SampleSettingTab](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L548) Rename these
> 
> [newPath = path.join(newPath, newName);](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L107) Just join the two with a slash `/`, no need to import `path`
> 
> [console.log('replace text', linkText, newLinkText);](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L115) Please avoid unnecessary logging. If you need it for debugging purposes, add a check to only log during development.
> 
> [.setName('Image Resize Mode')](https://github.com/xRyul/obsidian-image-converter/blob/e14dcbe0aa9a25a26e7a664949fdacb37d8d9f42/main.ts#L591) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

Removed the path import and joined it instead via the **/** (many thanks for this suggestion! ), renamed the default names, removed the logs, replaced text with sentence case in UI

liamcain commented:
- [style.fontSize](https://github.com/xRyul/obsidian-image-converter/blob/dbcc8ec4842c98e71d745394cf7c40fc0f28da97/main.ts#L1032) You should avoid assigning styles via javascript. Instead, just add a class to the element and move all these styles into CSS so that they are more easily adaptable by themes.
- [getActiveEditor](https://github.com/xRyul/obsidian-image-converter/blob/dbcc8ec4842c98e71d745394cf7c40fc0f28da97/main.ts#L559) you should use [`activeEditor.editor`](https://github.com/obsidianmd/obsidian-api/blob/38dd22168d2925086371bfc59e36fd9121527a39/obsidian.d.ts#L4009)
- You have 2 main.ts files in your repo. Looks like one is a duplicate. I suggest fixing that so you don't get confused down the line.
